WebApp-OpenIDConnect-DotNet/OpenIdConnectOptionsSetup.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+
+namespace WebApp_OpenIDConnect_DotNet
+{
+    public class OpenIdConnectOptionsSetup : IConfigureOptions<OpenIdConnectOptions>
+    {
+        public OpenIdConnectOptionsSetup(IOptions<AzureAdB2COptions> b2cOptions)
+        {
+            AzureAdB2COptions = b2cOptions.Value;
+        }
+
+        public AzureAdB2COptions AzureAdB2COptions { get; set; }
+
+        public void Configure(OpenIdConnectOptions options)
+        {
+            options.ClientId = AzureAdB2COptions.ClientId;
+            options.Authority = AzureAdB2COptions.Authority;
+            
+
+            // TODO: Is this needed? Is the default sufficient?
+            options.TokenValidationParameters = new TokenValidationParameters() { NameClaimType = "name" };
+
+            options.Events = new OpenIdConnectEvents()
+            {
+                OnRedirectToIdentityProvider = OnRedirectToIdentityProvider,
+                OnRemoteFailure = OnRemoteFailure
+            };
+        }
+
+        public async Task OnRedirectToIdentityProvider(RedirectContext context)
```

Comment:
This guy throws a warning about it being async with no await.
Remove the await and add a return Task.FromResult(0); at the end of the method?